File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03042/p03042_num1_parsed.py
# Program

S = input()

month = int(S[:2])
year = int(S[2:])

if 1 <= month <= 12 and 0 <= year <= 99:
    if (month >= 1 and month <= 12) and (year >= 0 and year <= 99):
        if (month >= 1 and month <= 12) and (year >= 0 and year <= 99):
            print("AMBIGUOUS")
        else:
            if (month >= 1 and month <= 12) and (year >= 0 and year <= 99):
                print("MMYY")
            else:
                if (month >= 1 and month <= 12) and (year >= 0 and year <= 99):
                    print("YYMM")
                else:
                    print("NA")
else:
    print("NA")

File: Datasets/TrickyBugs/GenProgs/tc_generated_progs_cpp/p03042/p03042_num1_parsed.cpp
#include <iostream>
using namespace std;

int main()
{
    string S;
    cin >> S;

    int YY = stoi(S.substr(0, 2));
    int MM = stoi(S.substr(2, 2));

    bool isYYMM = false;
    bool isMMYY = false;

    if (YY >= 1 && YY <= 12)
    {
        isYYMM = true;
    }

    if (MM >= 1 && MM <= 12)
    {
        isMMYY = true;
    }

    if (isYYMM && isMMYY)
    {
        cout << "AMBIGUOUS";
    }
    else if (isYYMM)
    {
        cout << "YYMM";
    }
    else if (isMMYY)
    {
        cout << "MMYY";
    }
    else
    {
        cout << "NA";
    }

    return 0;
}
